lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -90,15 +95,12 @@ struct IMDeadCodeElimPass : public IMDeadCodeElimBase<IMDeadCodeElimPass> {
 };
 } // namespace
 
-void IMDeadCodeElimPass::markWireOrRegOrNode(Operation *op) {
-  assert(isWireOrRegOrNode(op) && "only a wire, a reg or a node is expected");
+void IMDeadCodeElimPass::markWireOrRegOrNodeOrMemory(Operation *op) {
+  assert(isWireOrRegOrMemoryNode(op) &&
+         "only a wire, a reg or a node is expected");
   if (!isDeletableWireOrRegOrNode(op))
```

Comment:
It would be good if this PR also renamed this method. I do think this is fine to just be `isDeletableComponent` or `isDeletable` as the method does no checking of the type.

Reply:
Renamed these to "isDeclaration" since that's what the ops have in common.